sklearn/neighbors/tests/test_tree.py

```diff
@@ -0,0 +1,96 @@
+# License: BSD 3 clause
+
+import pickle
+import itertools
+
+import numpy as np
+import pytest
+
+from sklearn.neighbors.dist_metrics import DistanceMetric
+from sklearn.neighbors.ball_tree import BallTree
+from sklearn.neighbors.kd_tree import KDTree
+
+from sklearn.utils import check_random_state
+from numpy.testing import assert_array_almost_equal
+
+rng = np.random.RandomState(42)
+V_mahalanobis = rng.rand(3, 3)
+V_mahalanobis = np.dot(V_mahalanobis, V_mahalanobis.T)
+
+DIMENSION = 3
+
+METRICS = {'euclidean': {},
+           'manhattan': {},
+           'minkowski': dict(p=3),
+           'chebyshev': {},
+           'seuclidean': dict(V=rng.random_sample(DIMENSION)),
+           'wminkowski': dict(p=3, w=rng.random_sample(DIMENSION)),
+           'mahalanobis': dict(V=V_mahalanobis)}
+
+KD_TREE_METRICS = ['euclidean', 'manhattan', 'chebyshev', 'minkowski']
+BALL_TREE_METRICS = list(METRICS.keys())
+
+
+def dist_func(x1, x2, p):
+    return np.sum((x1 - x2) ** p) ** (1. / p)
+
+
+def brute_force_neighbors(X, Y, k, metric, **kwargs):
+    D = DistanceMetric.get_metric(metric, **kwargs).pairwise(Y, X)
+    ind = np.argsort(D, axis=1)[:, :k]
+    dist = D[np.arange(Y.shape[0])[:, None], ind]
+    return dist, ind
+
+
+@pytest.mark.parametrize(
+        'Cls, metric',
+        itertools.chain(
+            [(KDTree, metric) for metric in KD_TREE_METRICS],
+            [(BallTree, metric) for metric in BALL_TREE_METRICS]))
+@pytest.mark.parametrize('k', (1, 3, 5))
+@pytest.mark.parametrize('dualtree', (True, False))
+@pytest.mark.parametrize('breadth_first', (True, False))
+def test_kd_tree_query(Cls, metric, k, dualtree, breadth_first):
```

Comment:
Could be renamed to `test_nn_tree_query` and the test file itself to `test_nn_tree.py` or even `test_nearest_neighbors_tree.py`.

Reply:
Sure, forgot to rename it -- fixed now.